decentraland/unity-renderer
# feat: ECS7 add PointerEventResult component

## Context
This PR adds Pointer Event results to send back to the scene.
We are deprecating pointer up/down and hovers since we are assuming that we want the pointer results for every collider on the Pointer input layer.
Pointer result component will be sent to the scene's root entity containing the last 30 pointer events

More info of the pointer events: decentraland/sdk#352

## Approach
`ECSPointerInputSystem` will process input (pointer down/up hover enter/exit) from `PointerEventController` (button down/up and raycast result. which will be stored in `DataStore_ECS7`) against pointer colliders and will add the new pointer event to internal component `InternalInputEventResults`
`ECSInputSenderSystem` will iterate through all recently modified `InternalInputEventResults` components and send the update `PBPointerEventsResult` component back to the scene

## How to test the changes?

Can be tested [here](https://decentraland.github.io/sdk-playground/?&renderer-branch=feat%2Fupdate-pointer-event-result-component&code=ZnVuY3Rpb24gY3JlYXRlQ3ViZSh4OiBudW1iZXIsIHk6IG51bWJlciwgejogbnVtYmVyLCBzcGF3bmVyID0gZmFsc2UpOiBFbnRpdHkgewogIGNvbnN0IGVudGl0eSA9IGVuZ2luZS5hZGRFbnRpdHkoKQoKICBUcmFuc2Zvcm0uY3JlYXRlKGVudGl0eSwgewogICAgcG9zaXRpb246IHsgeCwgeSwgeiB9LAogICAgc2NhbGU6IHsgeDogMSwgeTogMSwgejogMSB9LAogICAgcm90YXRpb246IHsgeDogMCwgeTogMCwgejogMCwgdzogMSB9CiAgfSkKCiAgTWVzaFJlbmRlcmVyLmNyZWF0ZShlbnRpdHksIHsgYm94OiB7IHV2czogW10gfSB9KQogIE1lc2hDb2xsaWRlci5jcmVhdGUoZW50aXR5LCB7IGJveDoge30gfSkKCiAgcmV0dXJuIGVudGl0eQp9CgpmdW5jdGlvbiBjaXJjdWxhclN5c3RlbShkdDogbnVtYmVyKSB7CiAgZm9yIChjb25zdCBbZW50aXR5XSBvZiBlbmdpbmUuZ2V0RW50aXRpZXNXaXRoKE1lc2hSZW5kZXJlciwgVHJhbnNmb3JtKSkgewogICAgY29uc3QgdHJhbnNmb3JtID0gVHJhbnNmb3JtLmdldE11dGFibGUoZW50aXR5KQogICAgdHJhbnNmb3JtLnJvdGF0aW9uID0gUXVhdGVybmlvbi5tdWx0aXBseSh0cmFuc2Zvcm0ucm90YXRpb24sIFF1YXRlcm5pb24uYW5nbGVBeGlzKGR0ICogMTAsIFZlY3RvcjMuVXAoKSkpCiAgfQp9CgpsZXQgbWF4VGltZXN0YW1wID0gMAoKZnVuY3Rpb24gc3Bhd25lclN5c3RlbSgpIHsKICBjb25zdCBjbGlja2VkQ3ViZXMgPSBlbmdpbmUuZ2V0RW50aXRpZXNXaXRoKE1lc2hSZW5kZXJlciwgTWVzaENvbGxpZGVyKQogIGZvciAoY29uc3QgW2VudGl0eV0gb2YgY2xpY2tlZEN1YmVzKSB7CiAgICBpZiAod2FzRW50aXR5Q2xpY2tlZChlbnRpdHksIEFjdGlvbkJ1dHRvbi5QT0lOVEVSKSkgewogICAgICBjcmVhdGVDdWJlKE1hdGgucmFuZG9tKCkgKiA4ICsgMSwgTWF0aC5yYW5kb20oKSAqIDgsIE1hdGgucmFuZG9tKCkgKiA4ICsgMSkKICAgIH0KICB9CiAgY29uc3QgcG9pbnRlclJlc3VsdHMgPSBQb2ludGVyRXZlbnRzUmVzdWx0LmdldE9yTnVsbChlbmdpbmUuUm9vdEVudGl0eSkKICBpZiAocG9pbnRlclJlc3VsdHMpIHsKICAgIGNvbnN0IG5ld0NvbW1hbmRzID0gcG9pbnRlclJlc3VsdHMuY29tbWFuZHMuZmlsdGVyKChpdGVtKSA9PiBpdGVtLnRpbWVzdGFtcCA%2BIG1heFRpbWVzdGFtcCkKICAgIGlmIChuZXdDb21tYW5kcy5sZW5ndGggPiAwKSB7CiAgICAgIGxvZyh7IG5ld0NvbW1hbmRzIH0pCiAgICAgIG1heFRpbWVzdGFtcCA9IE1hdGgubWF4KC4uLm5ld0NvbW1hbmRzLm1hcCgoaXRlbSkgPT4gaXRlbS50aW1lc3RhbXApKQogICAgfQogIH0KfQoKY3JlYXRlQ3ViZSg4LCAxLCA4LCB0cnVlKQplbmdpbmUuYWRkU3lzdGVtKGNpcmN1bGFyU3lzdGVtKQplbmdpbmUuYWRkU3lzdGVtKHNwYXduZXJTeXN0ZW0pCg%3D%3D) (Press RUN button after renderer finishes loading)
Clicking on the cube on the center of the scene should spawn more cubes
